spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/support/tagsprovider/GatewayPathTagsProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.support.tagsprovider;
+
+import io.micrometer.core.instrument.Tags;
+
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.web.server.ServerWebExchange;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
+
+/**
+ * @author Marta Medio
+ * @author Alberto C. Ríos
+ */
+public class GatewayPathTagsProvider implements GatewayTagsProvider {
+
+	private static final String START_PATH_PATTERN = "Paths: [";
+
+	private static final String END_PATH_PATTERN = "], match";
+
+	@Override
+	public Tags apply(ServerWebExchange exchange) {
+		Route route = exchange.getAttribute(GATEWAY_ROUTE_ATTR);
+
+		if (route != null) {
+			String rawPredicate = route.getPredicate().toString();
```

Comment:
I'm not sure what does `getPredicate` return but would it be possible to use one/more of its getters (if any) and build the value from them instead of parsing `toString`?

Reply:
> Providing an enable/disable mechanism for tags is a slippery slope. The issue with this is that users can configure it and they can end up applications where this is enabled and apps where this is not. These apps can ship metrics to the same backend and aggregating and visualizing metrics where the tags are not aligned is problematic.

I am not sure I agree that it is a slippery slope. I don't think it's any different to any other feature that one can enable or disable. The motivation for having it disabled was because I thought the original implementation was bound to fail due it relying on toString(). I was also nervous about the possibility of having a high cardinality...although I don't think it can be any more than the number of routes defined.